plugins/ocm-backend/src/helpers/config.ts

```diff
@@ -82,3 +85,18 @@ export const readOcmConfigs = (config: Config): OcmConfig[] => {
     .keys()
     .map(id => getHubClusterFromConfig(id, ocmConfigs.getConfig(id), config));
 };
+
+export const getOwnerFromHub = (hubConfig: Config): string => {
+  if (!hubConfig.keys().includes(OWNER_KEY)) {
+    return 'unknown';
+  }
+
+  if (typeof hubConfig.get(OWNER_KEY) === 'string') {
+    return hubConfig.getString(OWNER_KEY);
+  }
+  const name = hubConfig.getString(`${OWNER_KEY}.name`);
+  const namespace =
+    hubConfig.getOptionalString(`${OWNER_KEY}.namespace`) || DEFAULT_NAMESPACE;
+  const kind = hubConfig.getOptionalString(`${OWNER_KEY}.kind`) || 'group';
+  return `${kind}:${namespace}/${name}`;
```

Comment:
Do we need to do this parsing? Why don't we return it as is and let Backstage parse it? `spec.owner` field can be a compound reference.

https://backstage.io/docs/features/software-catalog/references#compound-references

We should avoid assuming defaults to avoid potentially diverting from upstream.

Reply:
The owner field in the spec can only be a string, so thats why I think we have to use this parser. However if the default values from the parser for `kind` and `namespace` are removed we don't assume any defaults and the backstage parser will fill in the defaults on its own so we avoid assuming the defaults.
```suggestion
  if (typeof hubConfig.get(OWNER_KEY) === 'string') {
    return hubConfig.getString(OWNER_KEY);
  }
  const name = hubConfig.getString(`${OWNER_KEY}.name`);
  const namespace =
    hubConfig.getOptionalString(`${OWNER_KEY}.namespace`);
  const kind = hubConfig.getOptionalString(`${OWNER_KEY}.kind`);
  return `${kind}:${namespace}/${name}`;
```

Reply:
@SamoKopecky let's do it so we allow the `OWNER_KEY` to be only a value that is already directly consumable by Backstage `.spec.owner`, WDYT?

Reply:
Yes that actually makes a lot more sense then creating just a new parser that basically does the same thing as the backstage owner parser. I'll change it to just a simple string value.

Reply:
The schema looks like this:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Config": {
      "additionalProperties": false,
      "properties": {
        "catalog": {
          "additionalProperties": false,
          "properties": {
            "providers": {
              "additionalProperties": false,
              "properties": {
                "ocm": {
                  "additionalProperties": {
                    "anyOf": [
                      {
                        "additionalProperties": false,
                        "properties": {
                          "kubernetesPluginRef": {
                            "description": "Match the cluster name in kubernetes plugin config",
                            "type": "string"
                          },
                          "owner": {
                            "description": "Owner reference to created cluster entities in the catalog",
                            "type": "string"
                          }
                        },
                        "required": [
                          "kubernetesPluginRef"
                        ],
                        "type": "object"
                      },
                      {
                        "additionalProperties": false,
                        "properties": {
                          "caData": {
                            "description": "Base64-encoded CA bundle in PEM format used for verifying the TLS cert presented by the API server",
                            "type": "string"
                          },
                          "name": {
                            "description": "Name that the hub cluster will assume in Backstage Catalog (in OCM this is always local-cluster which can be confusing)",
                            "type": "string"
                          },
                          "owner": {
                            "description": "Owner reference to created cluster entities in the catalog",
                            "type": "string"
                          },
                          "serviceAccountToken": {
                            "description": "Service Account Token which is used for querying data from the hub",
                            "type": "string"
                          },
                          "skipTLSVerify": {
                            "description": "Skip TLS certificate verification presented by the API server, defaults to false",
                            "type": "boolean"
                          },
                          "url": {
                            "description": "Url of the hub cluster API endpoint",
                            "type": "string"
                          }
                        },
                        "required": [
                          "name",
                          "url"
                        ],
                        "type": "object"
                      }
                    ],
                    "description": "Key is reflected as provider ID. Defines and claims plugin instance ownership of entities"
                  },
                  "type": "object"
                }
              },
              "type": "object"
            }
          },
          "type": "object"
        }
      },
      "type": "object"
    },
    "HubClusterConfig": {
      "additionalProperties": false,
      "properties": {
        "caData": {
          "description": "Base64-encoded CA bundle in PEM format used for verifying the TLS cert presented by the API server",
          "type": "string"
        },
        "name": {
          "description": "Name that the hub cluster will assume in Backstage Catalog (in OCM this is always local-cluster which can be confusing)",
          "type": "string"
        },
        "serviceAccountToken": {
          "description": "Service Account Token which is used for querying data from the hub",
          "type": "string"
        },
        "skipTLSVerify": {
          "description": "Skip TLS certificate verification presented by the API server, defaults to false",
          "type": "boolean"
        },
        "url": {
          "description": "Url of the hub cluster API endpoint",
          "type": "string"
        }
      },
      "required": [
        "url",
        "name"
      ],
      "type": "object"
    },
    "KubernetesPluginRef": {
      "additionalProperties": false,
      "properties": {
        "kubernetesPluginRef": {
          "description": "Match the cluster name in kubernetes plugin config",
          "type": "string"
        }
      },
      "required": [
        "kubernetesPluginRef"
      ],
      "type": "object"
    }
  }
}

```